python/py-crossbar/Portfile

```diff
@@ -0,0 +1,188 @@
+                        service.
+    long_description    {*}${description}
+
+    depends_build
+    depends_lib         port:py38-crossbar
+
+    set cb_config_path  ${prefix}/etc/${name}
+    set cb_data_path    ${prefix}/var/db/${name}
+    set cb_log_path     ${prefix}/var/log/${name}
+    set cb_share_path   ${prefix}/share/${name}
+
+    set cb_plist_path   ${prefix}/etc/LaunchDaemons/org.macports.${name}
+
+    set cb_config_file  ${cb_config_path}/config.json
+    set cb_config_src   ${cb_share_path}/config.json.example
+    set cb_log_file     ${cb_log_path}/crossbar.log
+
+    set cb_plist        org.macports.${name}.plist
+
+    set cb_bin          ${prefix}/bin/crossbar-3.8
+
+    set cb_user         _crossbar
+
+    add_users           ${cb_user} \
+                        group=${cb_user} \
+                        realname=Crossbar.io
+
+    distfiles
+
+    build {}
+
+    extract {
+        copy ${filespath}/${cb_plist} ${workpath}/
+    }
+
+    patch {
+        set cb_plist_template ${workpath}/${cb_plist}
+
+        reinplace "s|@NAME@|${name}|g"                  ${cb_plist_template}
+        reinplace "s|@USER@|${cb_user}|g"               ${cb_plist_template}
+        reinplace "s|@GROUP@|${cb_user}|g"              ${cb_plist_template}
+        reinplace "s|@DATADIR@|${cb_data_path}|g"       ${cb_plist_template}
+        reinplace "s|@BIN@|${cb_bin}|g"                 ${cb_plist_template}
+        reinplace "s|@CONF_FILE@|${cb_config_file}|g"   ${cb_plist_template}
+        reinplace "s|@LOGFILE@|${cb_log_file}|g"        ${cb_plist_template}
+    }
+
+    destroot.keepdirs-append \
+                        ${destroot}/${cb_config_path} \
+                        ${destroot}/${cb_data_path} \
+                        ${destroot}/${cb_log_path}
+
+    destroot {
+        xinstall -d -m 755                  ${destroot}${cb_config_path}
+        xinstall -d -m 755 -o ${cb_user}    ${destroot}${cb_data_path}
+        xinstall -d -m 755                  ${destroot}${cb_log_path}
+        xinstall -d -m 755                  ${destroot}${cb_share_path}
+
+        copy ${filespath}/config.json.example ${destroot}${cb_share_path}/
+
+        xinstall -d -m 755 ${destroot}${cb_plist_path}
+
+        xinstall -m 0644 -o root -W ${workpath} ${cb_plist} \
+            ${destroot}${cb_plist_path}
+
+        xinstall -d -m 755 ${destroot}/Library/LaunchDaemons
+
+        ln -s ${cb_plist_path}/${cb_plist} \
+            ${destroot}/Library/LaunchDaemons/${cb_plist}
+    }
+
+    post-activate {
+        # Copy the example configuration into place if there doesn't currently
+        # exist a config file.
+        if {![file exists ${cb_config_file}]} {
+            copy ${cb_config_src} ${cb_config_file}
+        }
+
+        # Create the log file if it doesn't exist already
+        if {![file exists ${cb_log_file}]} {
+            system -W ${prefix} "touch ${cb_log_file}"
```

Comment:
Just use:
```
touch ${cb_log_file}
```
No need to use `system` to start a new shell just to run the `touch` executable; we have Tcl function for that.